ipaplatform/__init__.py.in

```diff
@@ -5,8 +5,58 @@
 '''
 Module containing platform-specific functionality for every platform.
 '''
+import sys
 
 NAME = "__PLATFORM__"
 
-# FIXME: too much cyclic dependencies
-# from __PLATFORM__ import paths, tasks, services
+_new_module = type(sys)
+
+
+class IpaPlatformImporter(object):
+    mod_aliases = ['constants', 'paths', 'services', 'tasks']
+
+    def __init__(self, platform=NAME):
+        self.aliases = {}
+        for alias in self.mod_aliases:
+            destmodname = 'ipaplatform.{}.{}'.format(platform, alias)
+            aliasmodname = 'ipaplatform.{}'.format(alias)
+            self.aliases[aliasmodname] = destmodname
+
+    def find_module(self, fullname, path=None):
+        if fullname in self.aliases:
+            return self
+        return None
+
+    def load_module(self, fullname):
+        try:
+            return sys.modules[fullname]
+        except KeyError:
+            pass
+        try:
+            dest = self.aliases[fullname]
+        except KeyError:
+            raise ImportError(fullname)
+        # load destination
+        __import__(dest)
+        mod = sys.modules[dest]
+        # create a new module and copy module content
+        newmod = _new_module(fullname)
+        newmod.__dict__.update(mod.__dict__)
+        newmod.__loader__ = self
+        # register module
+        sys.modules[fullname] = newmod
+        return newmod
+
+    def is_package(self, fullname):
+        mod = self.load_module(fullname)
+        return hasattr(mod, '__path__')
+
+    def get_code(self, fullname):
+        self.load_module(fullname)
+        return None
+
+    def get_source(self, fullname):
+        self.load_module(fullname)
+        return None
+
+sys.meta_path.append(IpaPlatformImporter())
```

Comment:
The behavior with `__path__` is exactly the same as with the symlinks, which work just fine.